### PR TITLE
HV always enables PTM on ptm root capable root port

### PR DESCRIPTION
**Pull request by rliu88**
This patch has 2 parts:
hv: during pci init, hv enables PTM on root port which is PTM root capable
dm: Add a sanity check whether PTM is enabled

### CODE REVIEW

**Comment by dongyaozu** [comment]
@JasonChenCJ can u review?

File: devicemodel/hw/pci/ptm.c [hunk 1]
```diff
@@ -175,11 +178,23 @@ int ptm_probe(struct vmctx *ctx, struct passthru_dev *ptdev, int *vrp_sec_bus)
...
-		*vrp_sec_bus = add_vroot_port(ctx, ptdev, rp, rp_ptm_offset);
+		/* hv is responsible to ensure that PTM is enabled on hw root port if
+		 * root port is PTM root-capable.  If PTM root is not enabled already in physical
+		 * root port before guest launch, guest OS can only enable it in root port's virtual
+		 * config space and PTM may not function as desired.
+		 */
+		cap = get_ptm_reg_value(rp, PCIR_PTM_CTRL);
+		if (!(cap & PCIM_PTM_CTRL_ENABLE) || !(cap & PCIM_PTM_CTRL_ROOT_SELECT)) {
+			pr_err("%s Warning: guest is not allowed to enable PTM on root port %x:%x.%x.\n",
+				__func__, rp->bus, rp->dev, rp->func);
```
Comment (JasonChenCJ) [inline devicemodel/hw/pci/ptm.c:189]: should we return error here? Physical PTM not enabled lead to no PTM support in guest..

**Comment by JasonChenCJ** [comment]
Just one comments, others are ok for me.

File: devicemodel/hw/pci/ptm.c (hunk 1 above)
Comment (rliu88) [reply devicemodel/hw/pci/ptm.c:190]: Just made the fix and pushed the updated patch

**Review by JasonChenCJ — APPROVED** [review]

**Pull request merged**